plugin/configurable.go

```diff
@@ -20,3 +22,13 @@ type Configurable interface {
 	// InitFromViper initializes this component with properties from spf13/viper.
 	InitFromViper(v *viper.Viper, logger *zap.Logger)
 }
+
+// Inheritable is an interface that can be implement by some storage implementations
+// to provide a way to inherit configuration settings from another factory.
+type Inheritable interface {
```

Comment:
nit: I would move these to plugin/storage, they are not directly related to "configurable", which is factory/storage-agnostic